### ISSUE TITLE
Migrate to use namespaces

### ISSUE THREAD

**Issue by yscik** [Automattic/WP-Job-Manager#2681]

**Comment by donnchawp** [comment]
> *This was generated by AI during triage.*

Closing this as superseded by #2976.

The "migrate to namespaces" direction is already happening organically — about 21 of 92 files in `includes/` declare `namespace WP_Job_Manager` (or sub-namespaces like `WP_Job_Manager\UI`), and all new code is born namespaced. What's missing is the infrastructure to make further migration cheap and BC-safe.

Rather than keep a vague mega-issue open, #2976 captures a concrete first slice: **add a composer classmap autoloader for the plugin's classes**. That unblocks any subsequent per-subsystem migration without requiring a big-bang rename. Future namespace work for specific subsystems can be filed as smaller, scoped issues as the work is actually planned (and with `class_alias()` shims where BC matters for add-ons).

Thanks @yscik for raising the direction.
